src/picnit.nit

```diff
@@ -141,30 +141,34 @@ class CommandInstall
 				abort
 			end
 
-			install_from_git(git_repo, package_id)
+			install_from_git(git_repo, package_id, version)
 		else
 			var name = package_id.git_name
 			if name != null and name != "." and not name.is_empty then
 				name = name.to_lower
-				install_from_git(package_id, name)
+				install_from_git(package_id, name, version)
 			else
 				print_error "Failed to infer the package name"
 				exit 1
 			end
 		end
 	end
 
-	private fun install_from_git(git_repo, name: String)
+	private fun install_from_git(git_repo, name: String, version: nullable String)
 	do
 		check_git
 
 		var target_dir = picnit_lib_dir / name
+		if version != null then target_dir += "=" + version
 		if target_dir.file_exists then
 			print_error "Already installed"
 			exit 1
 		end
 
-		var cmd = "git clone {git_repo.escape_to_sh} {target_dir.escape_to_sh}"
+		var cmd_branch = ""
+		if version != null then cmd_branch = "--branch '{version}'"
+
+		var cmd = "git clone --depth 1 {cmd_branch} {git_repo.escape_to_sh} {target_dir.escape_to_sh}"
```

Comment:
if you do it in 2 steps `git clone` then `git checkout` then you can also use sha1, or better what `git describe --always` gives, as a version identifier

Reply:
Good idea. We would lose the "--depth 1", but we could add it as a second try if a direct clone fails.